src/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component } from 'react';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+
+import { InputGroup, Button } from '../../index';
+import AutoCompleteInput from './AutoCompleteInput';
+import AutoCompleteItems from './AutoCompleteItems';
+import { KEY_CODES } from '../../common/helpers';
+import { getActiveItems } from './helpers';
+
+class AutoComplete extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: ''
+    };
+  }
+
+  handleStateChange = ({ inputValue, ...rest }) => {
+    if (typeof inputValue === 'string') {
+      this.props.onInputUpdate(inputValue);
+      this.setState({ inputValue });
+    }
+  };
+
+  render() {
+    const {
+      onSearch,
+      labelText,
+      onInputUpdate,
+      items,
+      actionText,
+      ...rest
+    } = this.props;
+
+    const activeItems = getActiveItems(items);
+
+    return (
+      <Downshift
+        onStateChange={this.handleStateChange}
+        defaultHighlightedIndex={0}
+        selectedItem={this.state.inputValue}
+        {...rest}
+        render={({
+          getInputProps,
+          getItemProps,
+          getLabelProps,
+          isOpen,
+          inputValue,
+          highlightedIndex,
+          getRootProps,
+          selectedItem,
+          selectItem
+        }) => {
+          const shouldShowItems = isOpen && items.length > 0;
+          const autoCompleteItemsProps = {
+            items,
+            highlightedIndex,
+            selectedItem,
+            getItemProps,
+            activeItems
+          };
+
+          return (
+            <div>
+              {labelText && <label {...getLabelProps()}>{labelText}</label>}
+              <InputGroup>
+                <AutoCompleteInput
+                  onKeyPress={e => {
```

Comment:
Can you move this function to be apart of the class like you did with `handleStateChange`?

Reply:
Unfortunately we don't have the render prop props (say that three times fast :) in a class function so we'd have to do some pass-throughs which I'm not a fan of.

Reply:
Cool, I understand your point now, it will actually be really messy...